libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -317,53 +313,60 @@ private static JToken EntityResultToInstanceJObject(string text, Result result)
-                    // add value
-                    JToken values;
-                    if (!entitiesResult.TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out values))
+                    if (recognizerResult.Entities == null)
                     {
-                        values = new JArray();
-                        entitiesResult[result.Label.Name] = values;
+                        recognizerResult.Entities = new JObject();
                     }
 
-                    ((JArray)values).Add(EntityResultToJObject(text, result));
-
-                    // get/create $instance
-                    JToken instanceRoot;
-                    if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
+                    var entitiesResult = recognizerResult.Entities;
+                    foreach (var result in results)
                     {
-                        instanceRoot = new JObject();
-                        recognizerResult.Entities["$instance"] = instanceRoot;
-                    }
+                        // add value
+                        JToken values;
+                        if (!entitiesResult.TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out values))
+                        {
+                            values = new JArray();
+                            entitiesResult[result.Label.Name] = values;
+                        }
 
-                    // add instanceData
-                    JToken instanceData;
-                    if (!((JObject)instanceRoot).TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out instanceData))
-                    {
-                        instanceData = new JArray();
-                        instanceRoot[result.Label.Name] = instanceData;
-                    }
+                        ((JArray)values).Add(EntityResultToJObject(text, result));
+
+                        // get/create $instance
+                        JToken instanceRoot;
+                        if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
+                        {
+                            instanceRoot = new JObject();
+                            recognizerResult.Entities["$instance"] = instanceRoot;
+                        }
 
-                    ((JArray)instanceData).Add(EntityResultToInstanceJObject(text, result));
+                        // add instanceData
+                        JToken instanceData;
+                        if (!((JObject)instanceRoot).TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out instanceData))
+                        {
+                            instanceData = new JArray();
+                            instanceRoot[result.Label.Name] = instanceData;
+                        }
+
+                        ((JArray)instanceData).Add(EntityResultToInstanceJObject(text, result));
```

Comment:
Same comment as above, are we 100% this is a JArray? If yes could you help me understand for future reference, why it is instanceData declared as JToken?

Reply:
@carlosscastro, following this previous precedent: https://github.com/microsoft/botbuilder-dotnet/blob/402bc02b4cbbd2f4ec359134640e99211367e4a5/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Recognizers/EntityRecognizers/EntityRecognizer.cs

Reply:
The fact that the same code is elsewhere does not make it correct unfortunately. However we can investigate this separately from the current PR, so let's table this one! Thanks for the pointer.